src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+                    {
+                        Log.Error(Globals.LogTag, "Error in Sensor Recorder Read synchronously CallBack");
+                    }
+
+                    DataReceived?.Invoke(null, e);
+                    Log.Debug(Globals.LogTag, "Recorder data callback recieved");
+                    return true;
+                };
+
+                int error = Interop.SensorRecoder.Read(type, QueryHandle, _sensorRecorderDataCallBack, IntPtr.Zero);
+
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in synchronously reading recorder");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.Read Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        private static void AsyncReader(int type, IntPtr QueryHandle, Interop.SensorRecoder.SensorRecorderDataCb Cb, IntPtr _userData)
+        {
+            int error = Interop.SensorRecoder.ReadAsync(type, QueryHandle, Cb, IntPtr.Zero);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in asynchronously reading recorder");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.ReadAsync Failed");
+            }
+        }
+
+        /// <summary>
+        /// Queries the recorded data asynchronously.
+        /// </summary>
+        /// <param name="type">
+        /// Sensor type.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public async Task ReadAsync(int type)
+        {
+            if (CheckQuery())
+            {
+                _sensorRecorderDataCallBack = (int _type, int data, int remains, int _error, Int64 _userData) =>
+                {
+                    RecorderDataEventArgs e = new RecorderDataEventArgs(_type, data, remains, _error, _userData);
+
+                    if (_error != (int)SensorError.None)
+                    {
+                        Log.Error(Globals.LogTag, "Error in Sensor Recorder Data aynchronously CallBack");
+                    }
+
+                    DataReceived?.Invoke(null, e);
+                    Log.Debug(Globals.LogTag, "Recorder data callback recieved");
+                    return true;
+                };
+
+                await Task.Run(() => AsyncReader(type, QueryHandle, _sensorRecorderDataCallBack, IntPtr.Zero)).ConfigureAwait(false);
+
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Starts to record a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to be recorded.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void Start(int type)
```

Comment:
Why sensor type is int type?

Reply:
Will it be fine if i replace it with enum SensorType ?

Reply:
No, SensorType is not enough, 
because Sensor could be more than one
I recommend to use `Sensor` object